server/src/resolvers/getVariantsResolver/getVariantsResolver.ts

```diff
@@ -65,19 +66,45 @@ const resolveVariantQuery = async (args: QueryInput): Promise<CombinedVariantQue
     }
   });
 
-  // once variants are merged, handle annotations
-  const caddAannotations = settled.find(
+  // filter data that are not in user requested assemblyId
+  const dataForLiftover = combinedResults.filter(v => v.variant.assemblyId !== assemblyId);
+  // filter data that are already in user requested assemlbyId
+  let dataForAnnotation = combinedResults.filter(v => {
+    if (resolveAssembly(v.variant.assemblyId) === assemblyId) {
+      v.variant.assemblyId = assemblyId;
+      v.variant.assemblyIdCurrent = assemblyId;
+      return true;
+    } else return false;
+  });
+  let unliftedVariants: VariantQueryDataResult[] = [];
+
+  // perform liftOver if needed
+  if (dataForLiftover.length) {
+    const liftoverResults = await liftover(dataForAnnotation, dataForLiftover, assemblyId);
+    unliftedVariants = liftoverResults.unliftedVariants;
```

Comment:
These 3 lines can be written as:

`{ unliftedVariants, dataForAnnotation, annotationPosition } = liftoverResults;`